### PR TITLE
Add regex validation to GoogleBasicProfile fields.

### LINKED ISSUE #11: Add regex validation to fields of User entity
Add regex validation for fields:
1. Email
2. Username
3. Nickname

Maybe there is some standard regex that we can use

### PR DESCRIPTION
**Pull request by rarques**

### CODE REVIEW

**Comment by rarques** [comment]
At the beginning of the development I put the validation of the fields inside the User domain entity, thinking that it was business logic to ensure a specific format of the fields inside the entity. Even though, in a way, it can be considered business related, I think it's more fitting to consider it "input validation" and thus, put this validation in the entry point of our app. The controller, where the data enters from the outside and can be corrupt.

I also wanted to put this validation separated, to be re-used in the future in other entry points (e.g. creating a User without google sign-in). But since this is not planned, I just put it in the controller. 
I added validation for the fields inside GoogleBasicProfile. That does not have a direct mapping with the fields of User entity.

What do you think? @mduenast

File: src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java [hunk 1]
```diff
@@ -20,14 +24,23 @@ public GoogleAuthRequest() {
...
         private String givenName;
         @NotNull
+        @Pattern(regexp = "^[a-zA-Z0-9-_\\s]+$",
+                message = "FamilyName has invalid format. Expected format: ^[a-zA-Z0-9-_\\s]+$")
         private String familyName;
         @NotNull
+        @URL
         private String imageUrl;
         @NotNull
+        @Email(regexp = "^[A-Za-z0-9+_.-]+@(.+)$",
```
Comment (mduenast) [inline src/main/java/com/garannodou/role/controller/GoogleAuthRequest.java]: ^[A-Za-z0-9+_.-]+@([A-Za-z0-9.])+$

File: src/test/java/com/garannodou/role/domain/UserTest.java [hunk 2]
```diff
@@ -10,14 +11,18 @@
 
 public class UserTest {
 
-    @Test(expected = IllegalStateException.class)
+    @Test
     public void testFailsWhenTheUserNameIsNull() {
-        User user = new User(null, "nickname", "user@email.com", URI.create("https://uri-example.com"));
+        Assertions.assertThrows(IllegalStateException.class, () -> {
```
Comment (mduenast) [inline src/test/java/com/garannodou/role/domain/UserTest.java]: import static

File: src/test/java/com/garannodou/role/domain/UserTest.java [hunk 3]
```diff
@@ -10,14 +11,18 @@
...
+            new User(null, "nickname", "user@email.com", URI.create("https://uri-example.com"));
+        });
     }
 
-    @Test(expected = IllegalStateException.class)
+    @Test
     public void testFailsWhenTheUserNameIsBlank() {
-        User user = new User(" ", "nickname", "user@email.com", URI.create("https://uri-example.com"));
+        Assertions.assertThrows(IllegalStateException.class, () -> {
```
Comment (mduenast) [inline src/test/java/com/garannodou/role/domain/UserTest.java]: same

File: src/test/java/com/garannodou/role/domain/UserTest.java [hunk 4]
```diff
@@ -32,13 +37,18 @@ public void testNickNameIsEqualToTheUserNameWhenTheNickNameIsBlank() {
         assertThat(user.getNickName(), is(equalTo(user.getUserName())));
     }
 
-    @Test(expected = IllegalStateException.class)
+    @Test
     public void testFailsWhenTheEmailIsNull() {
-        User user = new User("username", "nickname", null, URI.create("https://uri-example.com"));
+        Assertions.assertThrows(IllegalStateException.class, () -> {
```
Comment (mduenast) [inline src/test/java/com/garannodou/role/domain/UserTest.java]: same

File: src/test/java/com/garannodou/role/domain/UserTest.java [hunk 5]
```diff
@@ -32,13 +37,18 @@ public void testNickNameIsEqualToTheUserNameWhenTheNickNameIsBlank() {
...
+            new User("username", "nickname", null, URI.create("https://uri-example.com"));
+        });
     }
 
-    @Test(expected = IllegalStateException.class)
+    @Test
     public void testFailsWhenTheEmailIsBlank() {
-        User user = new User("username", "nickname", " ", URI.create("https://uri-example.com"));
+        Assertions.assertThrows(IllegalStateException.class, () -> {
```
Comment (mduenast) [inline src/test/java/com/garannodou/role/domain/UserTest.java]: same

**Review by mduenast — APPROVED** [review]

**Pull request merged**